brightener.h

```diff
@@ -1,11 +1,44 @@
 #include <cstdint>
 #include "image.h"
 
-class ImageBrightener {
-    private:
-        std::shared_ptr<Image> m_inputImage;
-    public:
-        ImageBrightener(std::shared_ptr<Image> inputImage);
-        int BrightenWholeImage();
-        bool AddBrighteningImage(std::shared_ptr<Image> imageToAdd, int& attenuatedCount);
-};
+namespace ImageProcessor
+{
+
+    void ExecutePixelRunner(std::shared_ptr<Image> inputImage, const std::shared_ptr<Image>& imageToAdd, int& attenuatedCount)
+    {
+        auto BrightnerFun = [&attenuatedCount](uint8_t pixelGrayscale, uint8_t pixelGrayscaleToAdd) {
+            uint8_t outputGrayscale = pixelGrayscale;
+            if (outputGrayscale + pixelGrayscaleToAdd > 255) {
+                ++attenuatedCount;
+                outputGrayscale = 255;
+            }
+            else {
+                outputGrayscale += pixelGrayscaleToAdd;
+            }
+
+            return outputGrayscale;
+        };
+
+        inputImage->pixelRunner(BrightnerFun, imageToAdd);
+    }
+
+    bool BrightenWholeImage(std::shared_ptr<Image> inputImage, int& attenuatedCount, std::shared_ptr<Image> imageToAdd = nullptr) {
```

Comment:
This function is `BrightenWholeImage` - which means the entire image is brightened. Why would it need an `imageToAdd`? The original code was just adding a fixed grayscale to every pixel in the image?

Reply:
Modified the code in latest commits